libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1209,31 +1209,24 @@ std::string YulUtilFunctions::resizeArrayFunction(ArrayType const& _type)
 string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
 {
 	string functionName = "resize_array_" + _type.identifier();
-	return m_functionCollector.createFunction(functionName, [&]() {
+	return m_functionCollector.createFunction(functionName, [&](vector<string>& _args, vector<string>&) {
+		_args = {"array", "newLen"};
 		return Whiskers(R"(
-			function <functionName>(array, newLen) {
-				if gt(newLen, <maxArrayLength>) {
-					<panic>()
-				}
```

Comment:
So the only change here, except using new `createFunction` interface is that this check is moved to `increaseSize` function